xla/backends/gpu/runtime/dynamic_slice_thunk.h

```diff
@@ -132,6 +135,22 @@ class DynamicSliceThunk : public Thunk {
 
   // A mapping from argument index to the base offset in the `offsets_allocs_`.
   std::vector<int64_t> offsets_allocs_base_;
+
+  // Extracted HloModules for computing dynamic offsets. The modules here are
+  // not used, this is solely for keeping the modules alive and maintain their
+  // ownership with the thunk. The pointers to these offsets are stored in the
```

Comment:
I don't really understand the comment about them being used. Presumably, we do use them somewhere? If not, why is it necessary to keep them alive?